service/raft/raft_group0.cc

```diff
@@ -1189,10 +1193,18 @@ static future<bool> wait_for_peers_to_enter_synchronize_state(
         }
 
         (void) [] (netw::messaging_service& ms, abort_source& as, gate::holder pause_shutdown,
-                   std::vector<gms::inet_address> current_config,
+                   raft::config_member_set current_members_set, const group0_members& members0,
                    lw_shared_ptr<std::unordered_set<gms::inet_address>> entered_synchronize,
                    lw_shared_ptr<bool> retry, ::tracker<bool> tracker) -> future<> {
-            co_await max_concurrent_for_each(current_config, max_concurrency, [&] (const gms::inet_address& node) -> future<> {
+            co_await max_concurrent_for_each(current_members_set, max_concurrency, [&] (const raft::config_member& member) -> future<> {
+                auto node_opt = members0.get_inet_addr(member);
+
+                if (!node_opt.has_value()) {
+                    co_return;
```

Comment:
What will happen if some nodes return `synchronize` and we won't be able to resolve IPs for the rest of them?

Reply:
Okay, now I see the problem. In this case, it will behave as if all node return `synchronize` and the function will return `true` instead of retrying. Is it enough to set `retry` to `true` if IP wasn't resolved?

Reply:
Yep, it should be enough. Then the behavior for that node will be the same as if we timed out trying to contact it.
Please also print a warning that we cannot resolve the IP of this node here in this branch. Something like:
```
upgrade_log.warn("wait_for_peers_to_enter_synchronize_state: cannot resolve the IP of {}", member);
```

Reply:
Done.